### DISCUSSION TITLE
Confusion About Filer Active Active Cross Cluster Continuous Synchronization

### DISCUSSION THREAD

**Discussion by homayoonsadeghiarkami** [seaweedfs/seaweedfs#2058] [Q&A]
Hi Mr. Lu 
I`m trying to understand which part of these filers synchronization is asynchronous !!
as you told in docs "at the beginning, it will bootstrap from the beginning of time, or resume from the last replication checkpoint. Later, it will just run continuously and persist checkpoints periodically."
I understand the architecture but which part is asynchronous? because as you said each filer has its own local change logs and from reading these logs we can replicate each filer synchronously...
these are my questions specifically:
1.  Which part of this process is asynchronous? does it replicate like an incremental arch?something like rsync ...
2. I know active-active or active-passive pattern in failover clustering but what is the meaning of these patterns in this architecture (replication)?
3. Where are the locations of local change log files for each filer by default and how can I change it because `-logdir` does not work for filer or other components like volume server and ...
Thanks a lot 👌

**Comment by chrislusf** [comment]
> from reading these logs we can replicate each filer synchronously...

Asynchronously.

> 1. Which part of this process is asynchronous? does it replicate like an incremental arch?something like rsync ...

Replicating files is asynchronous. Not incremental.

> 2. I know active-active or active-passive pattern in failover clustering but what is the meaning of these patterns in this architecture (replication)?

Files changed in either cluster will show up in the other cluster.

> 3. Where are the locations of local change log files for each filer by default and how can I change it because `-logdir` does not work for filer or other components like volume server and ...

On each filer themselves, `http://localhost:8888/topics/.system/log/`

**Reply by homayoonsadeghiarkami** [reply]
Thanks a lot 
For Q3:
Is it Good To backup these change logs and how? How can I restore these files in disaster state for speeding up my process? Should I Use it this way or you implemented another way?

**Reply by chrislusf** [reply]
use `weed filer.meta.backup`

**Reply by homayoonsadeghiarkami** [reply]
Perfect thanks and how should I restore these backups? where are these change logs in disk or ram or generated by helping filer store? I`m using MySQL as filer store BTW …

**Reply by chrislusf** [reply]
use `weed filer.meta.backup` to another mysql and switch mysql when needed.

**Reply by homayoonsadeghiarkami** [reply]
Thanks But Where are these filer metadata change events which are in topics folder? are they in disk or ram? I think they are in ram ...

**Reply by chrislusf** [reply]
they are saved in SeaweedFS

**Reply by homayoonsadeghiarkami** [reply]
hmmm I get it now. 
thanks as always Perfect 👍

**Reply by homayoonsadeghiarkami** [reply]
I tried to log into a directory for further usage in my containers but it does not work
like this:

```
'master')
  	ARGS="-mdir=/data -volumePreallocate=false -volumeSizeLimitMB=512"
  	shift
  	exec /usr/bin/weed -v=4 -logdir=/var/log/seaweedfs/master -alsologtostderr=true master $ARGS $@
	;;
```
@chrislusf
